### PR TITLE
fix/basicQuery deductive closre

### PR DESCRIPTION
**Pull request by jeswr**
- chore: update example
- fix: return derivations from basicQuery by default

### CODE REVIEW

**Comment by jeswr** [comment]
@smessie can I quickly get your review in this; it fixes a breaking change that was accidentally introduced in the release made yesterday

In particular the test

```ts
expect(
      basicQuery(dataQuads),
    ).resolves.toBeRdfIsomorphic([DataFactory.quad(
      DataFactory.namedNode('http://example.org/socrates#Socrates'),
      DataFactory.namedNode('http://www.w3.org/1999/02/22-rdf-syntax-ns#type'),
      DataFactory.namedNode('http://example.org/socrates#Mortal'),
    )]));
```
as changed to;

```ts
expect(
      basicQuery(dataQuads),
    ).resolves.toBeRdfIsomorphic([]));
```

this reverts that change whilst keeping the same behavior for `n3reasoner`

File: lib/index.ts [hunk 1]
```diff
@@ -24,6 +24,39 @@ export { default as EYE_PVM } from './eye';
...
+export function n3reasoner(
+  data: Quad[] | string,
+  query?: Quad[] | string | undefined,
+  options?: IQueryOptions,
+): Promise<Quad[] | string>
+export function n3reasoner(
+  data: Quad[] | string,
+  query?: Quad[] | string | undefined,
+  options?: IQueryOptions,
+): Promise<Quad[] | string> {
```
Comment (smessie) [inline lib/index.ts:56]: Im probably overlooking something, but what's the difference between those 2?

File: lib/index.ts [hunk 2]
```diff
@@ -52,9 +85,7 @@ export function basicQuery(
 export function basicQuery(
   data: Quad[] | string,
   query?: Quad[] | string | undefined,
-  options?: IQueryOptions,
+  options: IQueryOptions = {},
 ): Promise<Quad[] | string> {
-  return executeBasicEyeQuery(SWIPL_BUNDLE, data, query, options);
+  return n3reasoner(data, query, { output: query ? undefined : 'deductive_closure', ...options });
```
Comment (smessie) [inline lib/index.ts:90]: Will this also work if `options` is undefined?
(On my phone so can't really test myself rn)

**Review by smessie — APPROVED** [review]

File: lib/index.ts (hunk 1 above)
Comment (jeswr) [reply lib/index.ts:56]: Nuance of type overloading: the bottom one is the internal function definition (or Implementation signature), everything above that are the possible type definitions that are exported to users (overload signature).

See https://dmitripavlutin.com/typescript-function-overloading/

File: lib/index.ts (hunk 2 above)
Comment (jeswr) [reply lib/index.ts:90]: Yes - and just pushed a test case for it

Comment (smessie) [reply lib/index.ts:90]: Great!

**Pull request merged**